web_advanced_search/static/src/js/control_panel/advanced_select_list_controller.js

```diff
@@ -3,8 +3,14 @@
  */
 odoo.define("web_advanced_search.select_create_list_controller", function (require) {
     "use strict";
+    require('web.config');                                                     
+    require('web.core');                                                         
+    require('web.Dialog');                                                     
+    require('web.dom');  
+    require('web.view_registry');  
     const controller_registry = require("web.select_create_controllers_registry");
     const selectCreateController = controller_registry.SelectCreateListController;
```

Comment:
@gfcapalbo I don't know why they chose this structure, but looking purely at the code, it seems that you need to require both the one with an underscore and the one without an underscore:

```
const controller_registry = require("web.select_create_controller_registry");
require("web._select_create_controllers_registry");
```

Because the second one with underscore is the one that extends the first one with `.SelectCreateListController` that you are looking to load, so you need to wait at least for that one to have executed before you can rely on the property to be there.

Reply:
[link](https://github.com/OCA/OCB/blob/14.0/addons/web/static/src/js/views/select_create_controllers_registry.js#L56)

Reply:
@thomaspaulb I have imported with and without underscore, but never both. That may be the solution?

Trying that out immediately.

Reply:
That should work, and then the non-underscored one should have the property set.

Reply:
@thomaspaulb Got it . I misread the scopes of the JS.
odoo.define('web._select_create_controllers_registry )    has what we need.


But when i included odoo.define('web._select_create_controllers_registry )   it did not work. 
WHat i don;t understand is why do we need also:


  ```
 odoo.define('web.select_create_controllers_registry', function (require) {              
   "use strict";                                                                           
                                                                                           
   return {};                                                                              
                                                                                           
   });                                                                                     
  
```

for this to work. It seems a noop  class, but somehow in the registry loaded is seems important. 
I could not find where/how.

Reply:
@gfcapalbo The technical explanation is simple: the one **without** underscore is defined empty, and the one **with** underscore adds content to the former.

Thus, the latter needs to be required to have the right timing (eg after that one is run) and the first needs to be required in order to actually be able to access the content that was added. Hence, both need requiring.

What this doesn't answer is the why of all of this, but Odoo must have had some reason.

Reply:
> 

Got it.